### ISSUE TITLE
Regularly appearing javascript error when using HTTP/2?

### ISSUE THREAD

**Issue by irgendwie** [nextcloud/richdocuments#96]
Sometimes when trying to open a file, the following happens and only a reload (or more than one reload) of the page helps (afterwards the file can be opened normally):

![2017-06-07-183513_1136x1193](https://user-images.githubusercontent.com/8360698/26890005-5066c8b4-4bb0-11e7-99b4-4ea946445e8b.png)

I'm using nginx 1.13.1 and have HTTP/2 enabled.
Collabora is just the normal docker container without modifications.
For the nginx configuration I used the config provided by @icewind1991 (https://icewind.nl/entry/collabora-online/), added the HTTP/2 feature and changed some stuff related to SSL/TLS.
(Yes I know nginx isn't officially supported, but nonetheless would be cool to maybe get some hints in order to fix the problem)

Nginx error log:
> [...]
2017/06/07 18:25:05 [error] 24516#24516: *4049 connect() failed (111: Connection refused) while connecting to upstream, client: %some_ip%, server: %some_dns%, request: "GET /loleaflet/52f0568e/branding.css HTTP/2.0", upstream: "https://[::1]:9980/loleaflet/52f0568e/branding.css", host: "%some_host%"
2017/06/07 18:25:06 [crit] 24516#24516: *4049 open() "/var/cache/nginx/proxy_temp/3/01/0000000013" failed (13: Permission denied) while reading upstream, client: %some_ip%, server: %some_dns%, request: "GET /loleaflet/52f0568e/bundle.js HTTP/2.0", upstream: "https://127.0.0.1:9980/loleaflet/52f0568e/bundle.js", host: "%some_dns%"

Nextcloud log (may be unrealted):
> [...]
{"reqId":"%some_id%","level":3,"time":"2017-06-07T16:25:03+00:00","remoteAddr":"%some_ip%","user":"%some_name%","app":"PHP","method":"GET","url":"\/apps\/richdocuments\/index?fileId=%some_id%&requesttoken=%some_token%","message":"Undefined variable: owneruid at \/srv\/nc\/apps\/richdocuments\/lib\/TokenManager.php#88","userAgent":"%some_agent%","version":"12.0.0.29"}

I'm not sure if this is an issue with the docker container or the richdocuments app, so feel free to close the issue here and redirect me to the issue tracker of the docker container.

**Comment by thebearon** [comment]
Is this something you started getting recently? The "Undefined variable: owneruid at /srv/nc/apps/richdocuments/lib/TokenManager.php#88" part of the log is issue #93.

**Comment by pranavk** [comment]
If you look at the <script> tags in loleaflet.html file (the main html file), you will see the sequence of scripts are 

bundle.js
w2ui.min.js (which is the top and bottom toolbar library)
toolbar.js (actual toolbar using w2ui.min.js)

Both 2nd and 3rd make use of jquery which is in bundle.js and it seems that in your case, by the time 2nd and 3rd are loaded, browser doesn't load jquery and gives undefined error. No idea why it is different in HTTP/2 but works in normal setup. As far as I know, the sequence of script tags determines the execution of scripts as well, so jquery should be loaded before w2ui and toolbar.js are executed.

**Comment by irgendwie** [comment]
@thebearon I was experiencing this issue since I started using richdocuments (2-3 version before the current release) - at least the undefined variable is unrelated as expected, good :+1:

@pranavk Basically this means it is not an issue with the richdocuments app, but with my configuration or other side effects?

I did look up the error and found the following on serverfault: https://serverfault.com/questions/235154/permission-denied-while-reading-upstream/683622#683622
In my cache files of nginx there were some files with owner root while others were owned by my nginx user.
I deleted the cache folder, restarted nginx and the error seems to be gone (since it wasn't appearing every time, I'm not 100% sure though but the permission denied error message of nignx seems reasonable for this kind of issue).

I'll reopen the issue if the error should occur again and sorry for the noise (if it was just a fail of my server admin skills). :speak_no_evil:
